lib/utils/parse/parse.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package parse
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// IsRoleVariable checks if the passed in string matches the variable pattern
+// {{external.foo}} or {{internal.bar}}. If it does, it returns the variable
+// prefix and the variable name. In the previous example this would be
+// "external" or "internal" for the variable prefix and "foo" or "bar" for the
+// variable name. If no variable pattern is found, trace.NotFound is returned.
+func IsRoleVariable(variable string) (string, string, error) {
+	// strip {{ and }} from the start and end of the variable
+	if !strings.HasPrefix(variable, "{{") || !strings.HasSuffix(variable, "}}") {
```

Comment:
so if currently someone includes `{{` with space in front or after, we will skip this without warning, right? I would do the following: 

* Trim all space before and after `{{` and `}}`
* If I will encounter `{{` or `}}` symbol in the role allowed logins, but incorrect syntax, I would make this a role validation error to help users troubleshoot variables.

Otherwise when someone does this:

```bash
allowed_logins: ["{{typo}"]
```

we will silently ignore, and it will be hard to debug.